pkg/cmd/release/download/download.go

```diff
@@ -0,0 +1,171 @@
+package download
+
+import (
+	"errors"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/internal/ghrepo"
+	"github.com/cli/cli/pkg/cmd/release/shared"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/spf13/cobra"
+)
+
+type DownloadOptions struct {
+	HttpClient func() (*http.Client, error)
+	IO         *iostreams.IOStreams
+	BaseRepo   func() (ghrepo.Interface, error)
+
+	TagName     string
+	FilePattern string
+	Destination string
+
+	// maximum number of simultaneous downloads
+	Concurrency int
+}
+
+func NewCmdDownload(f *cmdutil.Factory, runF func(*DownloadOptions) error) *cobra.Command {
+	opts := &DownloadOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "download <tag> [<pattern>]",
+		Short: "Download release assets",
+		Args:  cobra.MinimumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// support `-R, --repo` override
+			opts.BaseRepo = f.BaseRepo
+
+			opts.TagName = args[0]
+			if len(args) > 1 {
+				opts.FilePattern = args[1]
+			}
+
+			opts.Concurrency = 5
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return downloadRun(opts)
+		},
+	}
+
+	cmd.Flags().StringVarP(&opts.Destination, "dir", "C", ".", "The directory to download files into")
```

Comment:
Good question! I guess I was biased towards `-C` because commands like `tar`, `make`, and `git` support it. Wget does `-P, --directory-prefix`. I will change to `-D` 👍